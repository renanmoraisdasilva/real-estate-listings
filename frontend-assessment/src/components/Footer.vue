<template>
  <footer>
    <nav>
      <div>
        <h1>Real Estate Listings</h1>
        <p>123 Main Street</p>
        <p>City, State ZIP</p>
      </div>
      <div>
        <h2>Contact Us</h2>
        <p>Phone: 555-555-5555</p>
        <p>Email: info@company.com</p>
      </div>
      <div>
        <h2>Follow Us</h2>
        <a href="#">Facebook</a>
        <a href="#">Twitter</a>
        <a href="#">Instagram</a>
      </div>
    </nav>
  </footer>
</template>

<script setup lang="ts"></script>

<style>
footer {
  box-shadow: rgba(0, 0, 0, 0.16) 0px -1px 0px;
  padding: 20px;
  display: flex;
  justify-content: center;
}
</style>
