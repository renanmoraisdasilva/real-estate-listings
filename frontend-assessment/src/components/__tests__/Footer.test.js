/**
 * @vitest-environment happy-dom
 */

import { describe, it, expect } from "vitest"
import { mount } from "@vue/test-utils"
import Footer from "../Footer.vue"

describe("Footer", () => {
  it("renders correctly", () => {
    const wrapper = mount(Footer)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find("footer").exists()).toBe(true)
  })
})
