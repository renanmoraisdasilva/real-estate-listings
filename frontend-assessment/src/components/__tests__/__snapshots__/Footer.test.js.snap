// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders correctly 1`] = `
"<footer>
  <nav>
    <div>
      <h1>Real Estate Listings</h1>
      <p>123 Main Street</p>
      <p>City, State ZIP</p>
    </div>
    <div>
      <h2>Contact Us</h2>
      <p>Phone: 555-555-5555</p>
      <p>Email: info@company.com</p>
    </div>
    <div>
      <h2>Follow Us</h2><a href=\\"#\\">Facebook</a><a href=\\"#\\">Twitter</a><a href=\\"#\\">Instagram</a>
    </div>
  </nav>
</footer>"
`;
